app/services/lists/my_modules_service.rb

```diff
@@ -70,7 +76,41 @@ def sort_records
         @records = @records.order(Arel.sql('COALESCE(my_modules.archived_on, my_modules.archived_on) ASC'))
       when 'archived_on_DESC'
         @records = @records.order(Arel.sql('COALESCE(my_modules.archived_on, my_modules.archived_on) DESC'))
+      when 'age_ASC'
+        @records = @records.order(:created_at)
+      when 'age_DESC'
+        @records = @records.order(created_at: :desc)
+      when 'status_ASC'
+        @records = @records.order(:my_module_status_id)
+      when 'status_DESC'
+        @records = @records.order(my_module_status_id: :desc)
+      when 'designated_ASC'
+        @records = @records.left_joins(:user_my_modules)
```

Comment:
Check if distinct is needed here to avoid duplicates caused by join.

In this case, there might be duplicates if there are multiple designated users.